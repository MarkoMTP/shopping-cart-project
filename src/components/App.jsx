import { Link, Outlet } from 'react-router-dom';
import style from '../styles/App.module.css';

export default function App() {
  return (
    <>
      <div className={style.navbar}>
        Buy Something
        <Link to="homepage">HomePage</Link>
      </div>

      <div className={style.mainContent}>
        <Outlet></Outlet>
      </div>
    </>
  );
}
