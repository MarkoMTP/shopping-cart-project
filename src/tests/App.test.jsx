import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import App from '../components/App';
import HomePage from '../components/Homepage'; // Ensure correct import path

describe('App component navigation', () => {
  it('renders the navbar correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('navigates to the homepage and displays content', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<App />}>
            <Route path="homepage" element={<HomePage />} />
          </Route>
        </Routes>
      </MemoryRouter>
    );

    // Verify initial content
    expect(screen.queryByText('Welcome')).toBeNull(); // Ensure 'Welcome' is not on the initial page

    // Click the 'HomePage' link
    fireEvent.click(screen.getByText('HomePage'));

    // Verify that the content from HomePage is displayed
    expect(screen.getByText('Welcome')).toBeInTheDocument();
    expect(screen.getByText('This is the HOME PAGE(fake)')).toBeInTheDocument();
    expect(screen.getByText('Its just for practice')).toBeInTheDocument();
  });
});
