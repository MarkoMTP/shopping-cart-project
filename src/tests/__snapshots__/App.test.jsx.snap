// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component navigation > renders the navbar correctly 1`] = `
<DocumentFragment>
  <div
    class="_navbar_61351a"
  >
    <p>
      Buy Something
    </p>
    <ul
      class="_links_61351a"
    >
      <a
        href="/homepage"
      >
        HomePage
      </a>
      <br />
      <a
        href="/shoppingpage"
      >
        Shop
      </a>
      <br />
      <span>
        %
      </span>
      <a
        href="/cart"
      >
        0
      </a>
    </ul>
  </div>
  <div
    class="_homePageClass_61351a"
  />
</DocumentFragment>
`;
